### PR TITLE
first-commits: Initialize AMI with base Windows Server 2022

### PR DESCRIPTION
**Pull request by itsmostafa**
Initialize project with a base windows image and a bootstrap winrm file for packer to be able to connect to the instance via WinRM.

## 🗣 Description ##
This PR only contains the base Windows Server 2022 image for initializing the project.
The Windows Administrator password for this AMI is set by AWS SSM Parameter Store.

## 💭 Motivation and context ##
This project can eventually be used to be able to create machine images of Windows Commando VM.

## 🧪 Testing ##
Need to have proper credentials set as repository secrets

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] *All* future TODOs are captured in issues, which are referenced in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] All new and existing tests pass.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # These owners will be the default owners for everything in the
 # repo. Unless a later match takes precedence, these owners will be
 # requested for review when someone opens a pull request.
-* @dav3r @felddy @jsf9k @mcdonnnj
+* @dav3r @felddy @jsf9k @mcdonnnj @itsmostafa @xlyk
```
Comment (dav3r) [inline .github/CODEOWNERS:6]: Alphabetize these:
```suggestion
* @dav3r @felddy @itsmostafa @jsf9k @mcdonnnj @xlyk
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -15,7 +15,7 @@ all of which should be in this repository.
 
 If you want to report a bug or request a new feature, the most direct
 method is to [create an
-issue](https://github.com/cisagov/windows-commando-vm-packer/issues) in this
+issue](https://github.com/cisagov/windows-commando-packer/issues) in this
```
Comment (dav3r) [inline CONTRIBUTING.md:18]: This should match the repo name:
```suggestion
issue](https://github.com/cisagov/windows-commando-vm-packer/issues) in this
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -25,7 +25,7 @@ one.
 ## Pull requests ##
 
 If you choose to [submit a pull
-request](https://github.com/cisagov/windows-commando-vm-packer/pulls), you will
+request](https://github.com/cisagov/windows-commando-packer/pulls), you will
```
Comment (dav3r) [inline CONTRIBUTING.md:28]: Same as above:
```suggestion
request](https://github.com/cisagov/windows-commando-vm-packer/pulls), you will
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -135,9 +135,9 @@ can create and configure the Python virtual environment with these
 commands:
 
 ```console
-cd windows-commando-vm-packer
-pyenv virtualenv <python_version_to_use> windows-commando-vm-packer
-pyenv local windows-commando-vm-packer
+cd windows-commando-packer
+pyenv virtualenv <python_version_to_use> windows-commando-packer
+pyenv local windows-commando-packer
```
Comment (dav3r) [inline CONTRIBUTING.md:140]: Same as above:
```suggestion
cd windows-commando-vm-packer
pyenv virtualenv <python_version_to_use> windows-commando-vm-packer
pyenv local windows-commando-vm-packer
```

File: terraform-build-user/backend.tf [hunk 5]
```diff
@@ -5,6 +5,6 @@ terraform {
     dynamodb_table = "terraform-state-lock"
     profile        = "cool-terraform-backend"
     region         = "us-east-1"
-    key            = "windows-commando-vm-packer/terraform-build-user.tfstate"
+    key            = "windows-commando-packer/terraform-build-user.tfstate"
```
Comment (dav3r) [inline terraform-build-user/backend.tf:8]: Match repo name:
```suggestion
    key            = "windows-commando-vm-packer/terraform-build-user.tfstate"
```

File: terraform-build-user/providers.tf [hunk 6]
```diff
@@ -10,7 +10,7 @@ locals {
 
   tags = {
     Team        = "CISA - Development"
-    Application = "windows-commando-vm-packer"
+    Application = "windows-commando-packer"
```
Comment (dav3r) [inline terraform-build-user/providers.tf:13]: Match repo name:
```suggestion
    Application = "windows-commando-vm-packer"
```

File: terraform-post-packer/main.tf [hunk 7]
```diff
@@ -24,12 +24,12 @@ locals {
   account_name_regex  = format("^env[[:digit:]]+ \\(%s\\)$", local.images_account_type)
 }
 
-# The IDs of all cisagov/windows-commando-vm-packer AMIs
+# The IDs of all cisagov/windows-commando-packer AMIs
```
Comment (dav3r) [inline terraform-post-packer/main.tf:27]: Match repo name:
```suggestion
# The IDs of all cisagov/windows-commando-vm-packer AMIs
```

File: terraform-post-packer/main.tf [hunk 8]
```diff
@@ -24,12 +24,12 @@ locals {
...
 data "aws_ami_ids" "historical_amis" {
   filter {
     name = "name"
     values = [
-      "example-hvm-*-x86_64-ebs",
+      "windows-server-2022-*-x86_64-ebs",
```
Comment (dav3r) [inline terraform-post-packer/main.tf:32]: Reminder that this AMI pattern should change once we actually start installing Commando tools on here.

Comment (dav3r) [inline terraform-post-packer/main.tf:32]: Reminder that this name should change once we actually start installing Commando tools on here.

File: src/packer.pkr.hcl [hunk 9]
```diff
@@ -46,32 +53,33 @@ variable "skip_create_ami" {
...
+source "amazon-ebs" "windows" {
   ami_block_device_mappings {
     delete_on_termination = true
     device_name           = "/dev/xvda"
     encrypted             = true
     volume_size           = 8
     volume_type           = "gp3"
   }
-  ami_name                    = "example-hvm-${local.timestamp}-x86_64-ebs"
+  ami_name                    = "windows-server-2022-${local.timestamp}-x86_64-ebs"
```
Comment (dav3r) [inline src/packer.pkr.hcl:77]: Reminder that the AMI name should change once we actually start installing Commando tools on here.

File: src/packer.pkr.hcl [hunk 10]
```diff
@@ -83,51 +91,48 @@ source "amazon-ebs" "example" {
...
+  source_ami         = data.amazon-ami.windows_server_2022.id
   subnet_filter {
     filters = {
       "tag:Name" = "AMI Build"
     }
   }
   tags = {
-    Application        = "Example"
-    Base_AMI_Name      = data.amazon-ami.debian_bullseye.name
+    Application        = "Windows Server 2022"
```
Comment (dav3r) [inline src/packer.pkr.hcl:101]: We should use this tag to distinguish between AMIs created with this repo and those created by `cisagov/windows-server-packer`:
```suggestion
    Application        = "Windows Commando VM"
```

**Review by dav3r — CHANGES_REQUESTED** [review]
This looks mostly good to me, but I noted a few things that should be fixed.  Also, please make sure that any late-breaking changes in https://github.com/cisagov/windows-server-packer/pull/10 get mirrored here.

**Comment by dav3r** [comment]
@itsmostafa - Can you please elaborate on why you closed out this PR?

**Comment by itsmostafa** [comment]
> @itsmostafa - Can you please elaborate on why you closed out this PR?

I went over this with Kyle on Friday, his last day, and was under the impression that this PR is very outdated and we should instead build on PR #21 . What are your thoughts?

**Pull request closed without merging**